analyses/SC_cd_2010/01_prep_SC_cd_2010.R

```diff
@@ -0,0 +1,91 @@
+###############################################################################
+# Download and prepare data for `SC_cd_2010` analysis
+# © ALARM Project, June 2022
+###############################################################################
+
+suppressMessages({
+    library(dplyr)
+    library(readr)
+    library(sf)
+    library(redist)
+    library(geomander)
+    library(cli)
+    library(here)
+    devtools::load_all() # load utilities
+})
+
+# Download necessary files for analysis -----
+cli_process_start("Downloading files for {.pkg SC_cd_2010}")
+
+path_data <- download_redistricting_file("SC", "data-raw/SC")
+
+# download the enacted plan.
+url <- "https://redistricting.lls.edu/wp-content/uploads/sc_2010_congress_2011-10-28_2021-12-31.zip"
+path_enacted <- "data-raw/SC/SC_enacted.zip"
+download(url, here(path_enacted))
+unzip(here(path_enacted), exdir = here(dirname(path_enacted), "SC_enacted"))
+file.remove(path_enacted)
+path_enacted <- "data-raw/SC/SC_enacted/H3992.shp"
+
+
+# If large, consider checking to see if these files exist before downloading
+
+cli_process_done()
+
+# Compile raw data into a final shapefile for analysis -----
+shp_path <- "data-out/SC_2010/shp_vtd.rds"
+perim_path <- "data-out/SC_2010/perim.rds"
+
+if (!file.exists(here(shp_path))) {
+    cli_process_start("Preparing {.strong SC} shapefile")
+    # read in redistricting data
+    sc_shp <- read_csv(here(path_data), col_types = cols(GEOID20 = "c")) %>%
+        join_vtd_shapefile() %>%
+        st_transform(EPSG$SC)  %>%
+        rename_with(function(x) gsub("[0-9.]", "", x), starts_with("GEOID"))
+
+    # add municipalities
+    d_muni <- make_from_baf("SC", "INCPLACE_CDP", "VTD")  %>%
+        mutate(GEOID = paste0(censable::match_fips("SC"), vtd)) %>%
+        select(-vtd)
+    d_cd <- make_from_baf("SC", "CD", "VTD")  %>%
+        transmute(GEOID = paste0(censable::match_fips("SC"), vtd),
+            cd_2010 = as.integer(cd))
```

Comment:
I think this should be called `cd_2000` as in `cd_2000 = as.integer(cd)`. A bit confusing, but the year after cd should be the year corresponding to the cycle. So the lines that was enacted in 2011 for the 2012 elections is called cd_2010. The lines that come from `make_from_baf` are lines before the redistricting, so they are 2000 cycle lines